### PR TITLE
Add debug friendly names depending on versioning

### PR DESCRIPTION
**Pull request by McGiverGim**
This is a better implementation of https://github.com/betaflight/blackbox-log-viewer/pull/506 that maintains backward compatibility when needed.

### CODE REVIEW

**Comment by McGiverGim** [comment]
Sonar complained about tabs not being spaces... I've modified that too, the file is not too big, including lines not modified by me, so this can difficult a little the review. Sorry.

File: js/flightlog_fields_presenter.js [hunk 1]
```diff
@@ -609,37 +625,37 @@ function FlightLogFieldPresenter() {
...
-					return value.toFixed(0) + "\u03BCS";
-				case 'BATTERY':
-					switch (fieldName) {
-						case 'debug[0]':
-							return value.toFixed(0);
-						default:
-							return (value/10).toFixed(1) + "V"
-					}	
+        if(flightLog) {
+            var debugModeName = DEBUG_MODE[flightLog.getSysConfig().debug_mode]; // convert to recognisable name
```
Comment (haslinghuis) [inline js/flightlog_fields_presenter.js:637]:
```suggestion
            const debugModeName = DEBUG_MODE[flightLog.getSysConfig().debug_mode]; // convert to recognisable name
```

File: js/flightlog_fields_presenter.js [hunk 2]
```diff
@@ -771,17 +787,17 @@ function FlightLogFieldPresenter() {
...
-		return "";
-	};
+        }
+        return "";
+    };
         
     FlightLogFieldPresenter.fieldNameToFriendly = function(fieldName, debugMode) {
         if (debugMode) {
-			if(fieldName.includes('debug')) {
+            if(fieldName.includes('debug')) {
```
Comment (haslinghuis) [inline js/flightlog_fields_presenter.js:796]: :space_invader: 
```suggestion
            if (fieldName.includes('debug')) {
```

File: js/flightlog_fields_presenter.js [hunk 3]
```diff
@@ -106,326 +106,342 @@ function FlightLogFieldPresenter() {
...
+            'debug[2]':'Altitude Adjustment',
+            'debug[3]':'Rescue State',
+        },
+    };
+
+    let DEBUG_FRIENDLY_FIELD_NAMES = null;
+
+    FlightLogFieldPresenter.adjustDebugDefsList = function(firmwareType, firmwareVersion) {
+
+        DEBUG_FRIENDLY_FIELD_NAMES = {...DEBUG_FRIENDLY_FIELD_NAMES_INITIAL};
```
Comment (haslinghuis) [inline js/flightlog_fields_presenter.js:432]: Fancy !!!

**Review by haslinghuis — CHANGES_REQUESTED** [review]

File: js/flightlog_fields_presenter.js [hunk 4]
```diff
@@ -609,37 +625,37 @@ function FlightLogFieldPresenter() {
...
-				case 'PIDLOOP': 
-					return value.toFixed(0) + "\u03BCS";
-				case 'BATTERY':
-					switch (fieldName) {
-						case 'debug[0]':
-							return value.toFixed(0);
-						default:
-							return (value/10).toFixed(1) + "V"
-					}	
+        if(flightLog) {
```
Comment (haslinghuis) [inline js/flightlog_fields_presenter.js:636]:
```suggestion
        if (flightLog) {
```

File: js/flightlog_fields_presenter.js (hunk 1 above)
Comment (McGiverGim) [reply js/flightlog_fields_presenter.js]: Done!

File: js/flightlog_fields_presenter.js (hunk 4 above)
Comment (McGiverGim) [reply js/flightlog_fields_presenter.js]: Done!

File: js/flightlog_fields_presenter.js (hunk 2 above)
Comment (McGiverGim) [reply js/flightlog_fields_presenter.js]: Done!

**Review by ctzsnooze — APPROVED** [review]
This is really fantastic.  Please merge!

**Review by mikeller — APPROVED** [review]

**Pull request merged**